### ISSUE TITLE
cc1plus: error: unrecognized command line option "-Wno-ignored-qualifiers"

### ISSUE THREAD

**Issue by dcalixto** [eventmachine/eventmachine#634]
Installing eventmachine 1.0.8 with native extensions

Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    /Users/danielcalixto/.rbenv/versions/2.2.2/bin/ruby -r ./siteconf20150902-42332-1kloes5.rb extconf.rb
checking for rb_trap_immediate in ruby.h,rubysig.h... no
checking for rb_thread_blocking_region()... no
checking for ruby/thread.h... yes
checking for rb_thread_call_without_gvl() in ruby/thread.h... yes
checking for inotify_init() in sys/inotify.h... no
checking for __NR_inotify_init in sys/syscall.h... no
checking for writev() in sys/uio.h... yes
checking for rb_thread_fd_select()... yes
checking for rb_fdset_t in ruby/intern.h... yes
checking for pipe2() in unistd.h... no
checking for accept4() in sys/socket.h... no
checking for SOCK_CLOEXEC in sys/socket.h... no
checking for rb_wait_for_single_fd()... yes
checking for rb_enable_interrupt()... no
checking for rb_time_new()... yes
checking for sys/event.h... yes
checking for sys/queue.h... yes
CFLAGS= -O3 -Wno-error=shorten-64-to-32  -pipe  -Wall -Wextra -Wno-deprecated-declarations -Wno-ignored-qualifiers -Wno-unused-result
CPPFLAGS=-I/Users/danielcalixto/.rbenv/versions/2.2.2/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT $(DEFS) $(cppflags) -Wall -Wextra -Wno-deprecated-declarations -Wno-ignored-qualifiers -Wno-unused-result
checking for clock_gettime()... no
checking for gethrtime()... no
creating Makefile

make "DESTDIR=" clean

make "DESTDIR="
compiling binder.cpp
cc1plus: error: unrecognized command line option "-Wno-ignored-qualifiers"
cc1plus: error: unrecognized command line option "-Wno-unused-result"
make: *** [binder.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/danielcalixto/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/eventmachine-1.0.8 for inspection.
Results logged to /Users/danielcalixto/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/extensions/x86_64-darwin-11/2.2.0-static/eventmachine-1.0.8/gem_make.out
An error occurred while installing eventmachine (1.0.8), and Bundler cannot continue.
Make sure that `gem install eventmachine -v '1.0.8'` succeeds before bundling.

**Comment by dcalixto** [comment]
someone same problem?

**Comment by sodabrew** [comment]
What are your compilers for CC and CXX / CPP ?

**Comment by dcalixto** [comment]
@sodabrew i'm using the command line tools for osx

**Comment by dcalixto** [comment]
right this is the version: xcode462_cltools_10_76938260a that i installed

**Comment by jackpipe** [comment]
Same problem here, eventmachine 1.0.8, mac os 10.8.5
> gcc --version
i686-apple-darwin11-llvm-gcc-4.2 (GCC) 4.2.1 (Based on Apple Inc. build 5658) (LLVM build 2336.11.00)

Just updating Xcode and cli tools to latest version, and will try again.

**Comment by jackpipe** [comment]
It now compiles, with the latest Xcode cli tools installed.
> gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 5.1 (clang-503.0.40) (based on LLVM 3.4svn)
Target: x86_64-apple-darwin12.6.0
Thread model: posix

**Comment by dcalixto** [comment]
right, one doubt @jackpipe, the latest Xcode that you said, by the way is the 6.4 version?

**Comment by jackpipe** [comment]
I'm on mac os 10.8.5, so the latest I can get is 5.1

**Comment by sodabrew** [comment]
Looks like the assumption that I can add the same flags to CPPFLAGS as CFLAGS is false. I'll work on a fix for the next release.

**Comment by dcalixto** [comment]
hah! so i'm fucked because i'm using mac os 10.7.5, and seems that the last Xcode compatible is 4.6. and i have any clue to to make a patch to solve this.

@sodabrew this problem is solved if i update to Xcode 4.6?

**Comment by yenshirak** [comment]
@dcalixto I installed GCC 5 with Homebrew (`brew install gcc`) then I created a symlink to  `g++-5` (`ln -s /usr/local/bin/g++-5 /usr/local/bin/g++`).

**Comment by pyrat** [comment]
Excellent thanks! Installing GCC 5 with homebrew fixed my problem!

**Comment by sodabrew** [comment]
@dcalixto Could you try #651 and let me know if that works?

**Comment by richardkmichael** [comment]
@sodabrew As no one replied to you, and I just had this problem myself, I thought I'd mention: #651 fixed it for me; OSX 10.8.5 with Xcode 4.6.3.  Specifically, I built from 076a526 (currently `master`).

I wonder if you'd consider releasing something now/soon; either `1.0.8.1` or changing the `1.0.9` milestone.  Now that Rails `>= 5.0.0.beta1` depends on eventmachine, while perhaps many edge Rails developers don't use an OS/compiler combo as old as mine, this issue might receive more traffic when additional Rails betas and especially final are released.  On one hand, it is perhaps a nuisance to be coerced by another project, but on the other, I guess there is about to be a large increase of eventmachine installs, and accompanying "issues".

In any case, thanks for the fix!

**Comment by sodabrew** [comment]
Thank you for the nudge, and for confirming the fix! I will put together a 1.0.9 release as soon as I can.

**Comment by ramonrails** [comment]
So, the issue started coming in :)
I was installing rails-5.0.0.beta1 and eventmachine install failed

```bash
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 7.0.2 (clang-700.1.81)
Target: x86_64-apple-darwin15.2.0
Thread model: posix
```

error at some compiler flags

```bash
make "DESTDIR="
compiling binder.cpp
couldn't understand kern.osversion `15.2.0'
cc1plus: error: unrecognized command line option "-Wno-ignored-qualifiers"
cc1plus: error: unrecognized command line option "-Wno-unused-result"
make: *** [binder.o] Error 1

make failed, exit code 2
```

I already attempted the g++ symlink step but it did not solve my issue.

* Mac OS X 10.11.2
* Xcode 7.2 (7C68)
* Command Line Tools OS X 10.11 for Xcode 7.2 (released on 2015-12-08)
* ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-darwin15]
* rbenv 0.4.0

I hope this helps. Looking forward to the fix. Thank you.

**Comment by sodabrew** [comment]
@ramonrails Using those same OS and CLT versions, I have no problems compiling eventmachine. Do you have an alternate GCC installed via HomeBrew by any chance?

**Comment by ramonrails** [comment]
I had. Uninstalled. Installed Xcode command line tools again. restarted machine. eventmachine still does not install.

—— error ----
compiling binder.cpp
couldn't understand kern.osversion `15.2.0'
cc1plus: error: unrecognized command line option "-Wno-ignored-qualifiers"
cc1plus: error: unrecognized command line option "-Wno-unused-result"
make: *** [binder.o] Error 1
----

**Comment by sodabrew** [comment]
I do have a fix for this pending in the master branch, which I am in process of picking out for a 1.0.9 release. Apologies for the delay on this.

**Comment by sodabrew** [comment]
Please try eventmachine-1.0.9 and let me know if it resolves this problem for you!

**Comment by ramonrails** [comment]
It does install now. Good help. Thank you @sodabrew 

```
 |2.2.3| MacBook-Pro in ~
○ → gem install eventmachine -v 1.0.9
Fetching: eventmachine-1.0.9.gem (100%)
Building native extensions.  This could take a while...
Successfully installed eventmachine-1.0.9
1 gem installed
```

**Comment by byrnejb** [comment]
Now I have this happening with 1.0.9 on OSX-10.9.5 with xcode 6.2. whereas 1.0.8 installed fine.  

<pre>
Installing eventmachine 1.0.9 with native extensions

Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    /opt/local/bin/ruby2.2 -r ./siteconf20160122-19000-1ylghh9.rb extconf.rb
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/opt/local/bin/$(RUBY_BASE_NAME)2.2
	--with-ssl-dir
	--without-ssl-dir
	--with-ssl-include
	--without-ssl-include=${ssl-dir}/include
	--with-ssl-lib
	--without-ssl-lib=${ssl-dir}/lib
	--with-openssl-config
	--without-openssl-config
	--with-pkg-config
	--without-pkg-config
/opt/local/lib/ruby2.2/2.2.0/mkmf.rb:456:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
	from /opt/local/lib/ruby2.2/2.2.0/mkmf.rb:541:in `try_link0'
	from /opt/local/lib/ruby2.2/2.2.0/mkmf.rb:556:in `try_link'
	from /opt/local/lib/ruby2.2/2.2.0/mkmf.rb:637:in `try_ldflags'
	from /opt/local/lib/ruby2.2/2.2.0/mkmf.rb:1780:in `pkg_config'
	from extconf.rb:64:in `pkg_config_wrapper'
	from extconf.rb:91:in `<main>'

extconf failed, exit code 1

Gem files will remain installed in /Users/byrnejb/Projects/hll_th/hll_th_proforma/libexec/bundle/lib/gems/eventmachine-1.0.9 for inspection.
Results logged to /Users/byrnejb/Projects/hll_th/hll_th_proforma/libexec/bundle/lib/extensions/x86_64-darwin-13/2.2.0/eventmachine-1.0.9/gem_make.out
</pre>

**Comment by sodabrew** [comment]
Please try event machine 1.0.9.1

**Comment by byrnejb** [comment]
On Fri, January 22, 2016 21:24, Aaron Stone wrote:
> Please try event machine 1.0.9.1
>

Actually I was able to get 1.0.9 to install once I updated my
MacPorts.  So, I do not know what it was choking on the first time. 
But, whatever it was it seems to have passed for now.


-- 
***          e-Mail is NOT a SECURE channel          ***
        Do NOT transmit sensitive data via e-Mail
James B. Byrne                mailto:ByrneJB@Harte-Lyne.ca
Harte & Lyne Limited          http://www.harte-lyne.ca
9 Brockley Drive              vox: +1 905 561 1241
Hamilton, Ontario             fax: +1 905 561 0757
Canada  L8E 3C3
